cpp/src/aggregators/sage.cu

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```

Comment:
@ChuckHastings @kaatish Should we better locate cugrpah ops related file in a separate directory? (e.g. src/ops_api?)

Reply:
Agree that a separate directory is probably better.
* `src/ops_api` could work
* `src/ops/aggregators` if we think we want more structure underneath might be a good option